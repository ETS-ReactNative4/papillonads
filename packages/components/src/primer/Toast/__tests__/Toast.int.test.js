import { renderCustom, renderInfo, renderSuccess, renderWarning, renderError, renderIsDismissable } from './Toast.int.render'

describe('<Button />', () => {
  const onClickMockFn = jest.fn()

  describe('Event', () => {
    test('must call onClick() when isDismissable', () => {
      const mountCustomRender = global.renderMount(renderCustom(onClickMockFn))
      mountCustomRender.find('button').first().simulate('click')
      expect(mountCustomRender.props().onClick).toBe(onClickMockFn)
      expect(mountCustomRender.props().onClick).toHaveBeenCalled()
    })
  })

  describe('Render', () => {
    test('must match renderInfo()', () => {
      expect(global.renderToJSON(renderInfo())).toMatchSnapshot()
    })

    test('must match renderSuccess()', () => {
      expect(global.renderToJSON(renderSuccess())).toMatchSnapshot()
    })

    test('must match renderWarning()', () => {
      expect(global.renderToJSON(renderWarning())).toMatchSnapshot()
    })

    test('must match renderError()', () => {
      expect(global.renderToJSON(renderError())).toMatchSnapshot()
    })

    test('must match renderIsDismissable()', () => {
      expect(global.renderToJSON(renderIsDismissable())).toMatchSnapshot()
    })
  })
})
