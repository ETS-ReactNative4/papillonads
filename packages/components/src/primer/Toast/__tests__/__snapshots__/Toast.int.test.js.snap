// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Render must match renderError() 1`] = `
<div
  className="p-1"
>
  <div
    className="Toast Toast--error"
  >
    <span
      className="Toast-icon"
    >
      <mock-file
        aria-hidden="true"
        aria-label={null}
        className="octicon"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      />
    </span>
    <span
      className="Toast-content"
    >
      Error message goes here
    </span>
  </div>
</div>
`;

exports[`<Button /> Render must match renderInfo() 1`] = `
<div
  className="p-1"
>
  <div
    className="Toast"
  >
    <span
      className="Toast-icon"
    >
      <mock-file
        aria-hidden="true"
        aria-label={null}
        className="octicon"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      />
    </span>
    <span
      className="Toast-content"
    >
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor.Aenean massa. Cum sociis natoquepenatibus et ma
    </span>
  </div>
</div>
`;

exports[`<Button /> Render must match renderIsDismissable() 1`] = `
<div
  className="p-1"
>
  <div
    className="Toast"
  >
    <span
      className="Toast-icon"
    >
      <mock-file
        aria-hidden="true"
        aria-label={null}
        className="octicon"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      />
    </span>
    <span
      className="Toast-content"
    >
      This toast is dismissable.
    </span>
    <button
      className="Toast-dismissButton"
      onClick={[Function]}
      type="button"
    >
      <mock-file
        aria-hidden="true"
        aria-label={null}
        className="octicon"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      />
    </button>
  </div>
</div>
`;

exports[`<Button /> Render must match renderSuccess() 1`] = `
<div
  className="p-1"
>
  <div
    className="Toast Toast--success"
  >
    <span
      className="Toast-icon"
    >
      <mock-file
        aria-hidden="true"
        aria-label={null}
        className="octicon"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      />
    </span>
    <span
      className="Toast-content"
    >
      Success message goes here
    </span>
  </div>
</div>
`;

exports[`<Button /> Render must match renderWarning() 1`] = `
<div
  className="p-1"
>
  <div
    className="Toast Toast--warning"
  >
    <span
      className="Toast-icon"
    >
      <mock-file
        aria-hidden="true"
        aria-label={null}
        className="octicon"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      />
    </span>
    <span
      className="Toast-content"
    >
      Warning message goes here
    </span>
  </div>
</div>
`;
